src/components/graph/graph.ts

```diff
@@ -422,32 +429,72 @@ export default class GraphComponent {
-        d3.select(refs[i]).attr('y1', sourceNode.y)
-        return sourceNode.x
-      })
-      .attr('x2', (l: ILinkTuple, i: number, refs: any[]) => {
-        const targetNode = this._graphData!.metadataItems.filter(
-          (d: IGraphNodeData) => d.id === l.target)[0]
-        d3.select(refs[i]).attr('y2', targetNode.y)
-        return targetNode.x
-      })
-      .attr('stroke-width', GraphComponent._LINK_STROKE)
-      .on('contextmenu', (l: ILinkTuple) => this._onLinkContextMenu(l))
 
+    const generateLink = {
+      vertical: d3.linkVertical()
+        .x((l: any) => {
+          return this._graphData!.metadataItems.filter(
+            (d: IGraphNodeData) => d.id === l)[0].x
+        })
+        .y((l: any) => {
+          return this._graphData!.metadataItems.filter(
+            (d: IGraphNodeData) => d.id === l)[0].y 
+        }),
+
+      horizontal: d3.linkHorizontal()
+        .x((l: any) => {
+          return this._graphData!.metadataItems.filter(
+            (d: IGraphNodeData) => d.id === l)[0].x
+        })
+        .y((l: any) => {
+          return this._graphData!.metadataItems.filter(
+            (d: IGraphNodeData) => d.id === l)[0].y
+        }),
+    }
+
+    switch (this._linkMode) {
+      case LinkType.STRAIGHT:
+        existingLinks
+          // NOTE: Key-function provides D3 with information about which datum maps to which
+          // element. This allows arrays in different orders to work as expected
+          .append('line')
+          .attr('class', 'link')
+          .attr('x1', (l: ILinkTuple, i: number, refs: any[]) => {
+            const sourceNode = this._graphData!.metadataItems.filter(
+              (d: IGraphNodeData) => d.id === l.source)[0]
+            d3.select(refs[i]).attr('y1', sourceNode.y)
+            return sourceNode.x
+          })
+          .attr('x2', (l: ILinkTuple, i: number, refs: any[]) => {
+            const targetNode = this._graphData!.metadataItems.filter(
+              (d: IGraphNodeData) => d.id === l.target)[0]
+            d3.select(refs[i]).attr('y2', targetNode.y)
+            return targetNode.x
+          })
+        break
+      case LinkType.VERTICAL: 
+        existingLinks.append('path')
+          .attr('class', 'link')
+          .attr('d', generateLink.vertical)
+        break
+      case LinkType.HORIZONTAL: 
+        existingLinks
+          .append('path')
+          .attr('class', 'link')
+          .attr('d', generateLink.horizontal)
+        break
+      default: break
+    }
+      
     existingLinks
       // NOTE: Key-function provides D3 with information about which datum maps to which
       // element. This allows arrays in different orders to work as expected
+      .selectAll('.link')
+      .attr('stroke-width', GraphComponent._LINK_STROKE)
+      .attr('fill', 'none')
+      .on('contextmenu', (l: ILinkTuple) => this._onLinkContextMenu(l))
       .data(this._graphData!.linkData, (l: ILinkTuple) => l.id)
```

Comment:
Can you move this up, it would make more sense a few lines above rather than at the very bottom. And the comment about the 'Key-function' is referring to this line, so move that comment to be next to this as well.